Add complete set of PDF.js context menu items

nt1m opened the pull request:
#### f45efb47d0b878c56267ea737cdd1d1bdb6a41e7
<pre>
Add complete set of PDF.js context menu items
<a href="https://bugs.webkit.org/show_bug.cgi?id=243709">https://bugs.webkit.org/show_bug.cgi?id=243709</a>
rdar://98356019

Reviewed by Aditya Keerthi.

- Add zoom-related items and next/previous page items
- Fix default zoom
- Hide Back/Forward/Inspect Element items
- Add leading separator when there is a preceding context menu item like Share/Services/etc.

* Source/WebCore/Modules/pdfjs-extras/content-script.js:
(const.PDFJSContentScript.overrideSettings):
(const.PDFJSContentScript.setPageMode):
(const.PDFJSContentScript.autoResize):
(const.PDFJSContentScript.init):
* Source/WebCore/en.lproj/Localizable.strings:
* Source/WebCore/html/PDFDocument.cpp:
(WebCore::PDFDocument::createDocumentStructure):
* Source/WebCore/page/ContextMenuController.cpp:
(WebCore::ContextMenuController::contextMenuItemSelected):
(WebCore::ContextMenuController::populate):
(WebCore::ContextMenuController::addInspectElementItem):
* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::contextMenuItemPDFAutoSize):
(WebCore::contextMenuItemPDFZoomIn):
(WebCore::contextMenuItemPDFZoomOut):
(WebCore::contextMenuItemPDFActualSize):
(WebCore::contextMenuItemPDFNextPage):
(WebCore::contextMenuItemPDFPreviousPage):
* Source/WebCore/platform/LocalizedStrings.h:

Canonical link: <a href="https://commits.webkit.org/253250@main">https://commits.webkit.org/253250@main</a>
</pre>

pxlcoder commented on Source/WebCore/Modules/pdfjs-extras/content-script.js:
```diff
@@ -27,18 +27,27 @@ const PDFJSContentScript = {
         delete PDFViewerApplication.supportsIntegratedFind;
         PDFViewerApplication.supportsIntegratedFind = true;
+
+        // Hide sidebar on load
```
Nit: `// Hide the sidebar on load.` (following [comment style guidelines](https://webkit.org/code-style-guidelines/#comments-sentences))

pxlcoder commented on Source/WebCore/Modules/pdfjs-extras/content-script.js:
```diff
@@ -27,18 +27,27 @@ const PDFJSContentScript = {
         // Disable the findbar provided by PDF.js
```
Nit: Punctuation here.

pxlcoder commented on Source/WebCore/page/ContextMenuController.cpp:
```diff
@@ -1235,6 +1270,12 @@ void ContextMenuController::addInspectElementItem()
 
+#if ENABLE(PDFJS)
+    RefPtr ownerElement = frame->ownerElement();
+    if (ownerElement && ownerElement->document().isPDFDocument())
```
Nit: `if (RefPtr ownerElement = frame->ownerElement(); ownerElement && ownerElement->document().isPDFDocument())`.

pxlcoder commented on Source/WebCore/page/ContextMenuController.cpp:
```diff
@@ -1058,10 +1083,20 @@ void ContextMenuController::populate()
+                appendItem(PDFTwoPagesContinuousItem, m_contextMenu.get());
+                appendItem(*separatorItem(), m_contextMenu.get());
+                appendItem(PDFNextPageItem, m_contextMenu.get());
+                appendItem(PDFPreviousPageItem, m_contextMenu.get());
```
Might be nice to have a newline between each section (separator) for readability.

pxlcoder commented on Source/WebCore/Modules/pdfjs-extras/content-script.js:
```diff
@@ -27,18 +27,27 @@ const PDFJSContentScript = {
+    },
+
+    autoResize() {
+        if (PDFViewerApplication.pdfViewer.spreadMode == 1) // Two pages
```
Are there named constants for `spreadMode` and `scrollMode`? Would be nice to use one if possible.

nt1m replied:
There are, but they're not exposed, unfortunately.

pxlcoder replied:
Can we redefine (since we're already relying on the integer value) or get them exposed in PDF.js?

nt1m committed:
Add complete set of PDF.js context menu items
https://bugs.webkit.org/show_bug.cgi?id=243709
rdar://98356019

Reviewed by Aditya Keerthi.

- Add zoom-related items and next/previous page items
- Fix default zoom
- Hide Back/Forward/Inspect Element items
- Add leading separator when there is a preceding context menu item like Share/Services/etc.

* Source/WebCore/Modules/pdfjs-extras/content-script.js:
(const.PDFJSContentScript.overrideSettings):
(const.PDFJSContentScript.setPageMode):
(const.PDFJSContentScript.autoResize):
(const.PDFJSContentScript.init):
* Source/WebCore/en.lproj/Localizable.strings:
* Source/WebCore/html/PDFDocument.cpp:
(WebCore::PDFDocument::createDocumentStructure):
* Source/WebCore/page/ContextMenuController.cpp:
(WebCore::ContextMenuController::contextMenuItemSelected):
(WebCore::ContextMenuController::populate):
(WebCore::ContextMenuController::addInspectElementItem):
* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::contextMenuItemPDFAutoSize):
(WebCore::contextMenuItemPDFZoomIn):
(WebCore::contextMenuItemPDFZoomOut):
(WebCore::contextMenuItemPDFActualSize):
(WebCore::contextMenuItemPDFNextPage):
(WebCore::contextMenuItemPDFPreviousPage):
* Source/WebCore/platform/LocalizedStrings.h:

Canonical link: https://commits.webkit.org/253250@main
